### PR TITLE
ELK stack preinstalled.

### PR DESCRIPTION
**Pull request by corebug**
ELK stack preinstalled for Mantl.

Uses `nginx-consul` on edge nodes as unified endpoint for ES (by now, until frameworks learn to query ES endpoints from Consul).

### CODE REVIEW

File: terraform.sample.yml [hunk 1]
```diff
@@ -16,6 +25,10 @@
     # this variable.
     # consul_acl_datacenter: your_primary_datacenter
 
+    # consul_dns_domain is repeated across these plays so that all the
+    # components know what information to use for this values to help with
+    # service discovery.
+    consul_dns_domain: consul
```
Comment (ChrisAubuchon) [inline terraform.sample.yml]: `consul_dns_domain` is set in `security-setup` and should be removed

File: terraform.sample.yml [hunk 2]
```diff
@@ -44,6 +47,7 @@
   # all servers, we can skip it here to speed up the deployment.
   gather_facts: no
   vars:
+    consul_dns_domain: consul
```
Comment (ChrisAubuchon) [inline terraform.sample.yml]: I missed this one

File: terraform.sample.yml [hunk 3]
```diff
@@ -56,6 +60,7 @@
 - hosts: role=control
   gather_facts: no
   vars:
+    consul_dns_domain: consul
```
Comment (ChrisAubuchon) [inline terraform.sample.yml]: And this one...

File: roles/mesos-elasticsearch/README.rst [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Elasticsearch database using the Mesos Elasticsearch framework
+
+Variables
+---------
+
+You can use these variables to customize your Elasticsearch installations:
+
+.. data :: example_option_name
+
+   Example text
```
Comment (ryane) [inline roles/mesos-elasticsearch/README.rst]: none of the variables are actually documented here

File: roles/mesos-kibana/README.rst [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+Kibana using the Mesos Kibana framework
+
+Variables
+---------
+
+You can use these variables to customize your Kibana installations:
+
+.. data :: example_option_name
+
+   Example text
```
Comment (ryane) [inline roles/mesos-kibana/README.rst]: needs documentation

File: roles/mesos-logstash/README.rst [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+Logstash using the Mesos Logstash framework
+
+Variables
+---------
+
+You can use these variables to customize your Logstash installations:
+
+.. data :: example_option_name
+
+   Example text
```
Comment (ryane) [inline roles/mesos-logstash/README.rst]: docs needed

**Comment by ryane** [comment]
I'm finding several issues with this PR:

* There are merge conflicts that need to be resolved.

* The nginx-es role fails with the following error: 

    ```
    fatal: [resching-aws-edge-02] => input file not found at /roles/consul/defaults/main.yml or /roles/consul/defaults/main.yml
    ```

* I am not sure why we moved away from using Docker within these frameworks but, right now, none of these will run on Mantl since Java is not installed on worker nodes. It looks like we'd need to install Java as a pre-requisite on all worker nodes.
    
* It seems like none of these frameworks (Elasticsearch, Logstash, or Kibana) are configured to work with Mesos authentication. Since Mantl clusters have Mesos authentication turned on by default, none of these frameworks will run on default Mantl installations.

* The number of Elasticsearch executor nodes is not configurable
    * I believe the default configuration will attempt to run 3 nodes. This means the framework will not have enough resources on Vagrant clusters nor on AWS/GCE clusters using the default sample configuration files.
    * If we are going to include this by default, it seems we either need to do some kind of dynamic configuration to determine the appropriate sizing or exclude it from clusters that don't meet the minimum system requirements. We should at least document the minimum system requirements as well as a recommended configuration for production clusters.

* What is the use case for the Logstash framework? We already install Logstash on all nodes. With this PR, it looks like we will attempt to install another instance of Logstash on all worker nodes. Logstash is very heavy so I don't think we want two instances running on each worker. Is it intended to replace the existing instances of Logstash? Does it preserve the same configuration? If so, we need to remove the existing Logstash role from worker nodes. But, what about control and edge nodes? Collecting logs from those machines is very important and I am not seeing any changes to configure those nodes to ship logs to Elasticsearch. Please let me know if I missed anything.

**Comment by zoza1982** [comment]
@ryane haven't you and @corebug synced on that ES can be installed via Mantl API which runs in container and it takes care of security too? ( Not sure about logstash and kibana framework)

I am confused a little due to duplicated efforts...

We should document which frameworks are being worked on and by whom..regardless of lack of official documentation on how to install/run them.

**Comment by ryane** [comment]
yes we have. 

The elasticsearch framework is the same; just a different approach to installation and configuration. The logstash and kibana frameworks are more recent. The intent is to tightly integrate the elk stack into mantl.

**Comment by siddharthist** [comment]
Fixed in #1127

**Pull request closed without merging**